README.example.md

```diff
@@ -51,32 +53,83 @@ This section should include any business related explanation that helps understa
 If the project is using some special external services (NemID, oAuth, Customer API, etc.) give a short description here.
 Example: The app is using NemID to authenticate all its users.
 
-# Requirements
+# Getting the project running
 
-You must have the following installed and available on your machine:
+There are three supported ways to run the project:
 
-- **Ruby 3.0.x**
-- **Node JS 16.x**
-- **Yarn 1.x**
-- **PostgreSQL 12**
-- **Redis**
+1. [Run locally using docker for database servers](#run-locally-using-docker-for-database-servers)
+2. [Run entirely project locally (for non-docker purists)](#run-entirely-project-locally)
+3. [Run inside docker dev container (for docker purists)](#run-inside-docker-dev-container)
 
-## Installing requirements
+## Run locally using docker for database servers
 
-With [asdf](https://asdf-vm.com/) and [docker](https://www.docker.com/) (or [podman](https://podman.io/)) you can easily install/run the required versions of runtimes and services.
+Requirements:
 
-Install runtimes with `asdf`:
+- docker (or podman)
+- asdf
 
-```
+1. First install node and ruby:
+
+```bash
+# Add nodejs and ruby plugins if you don't already have them
+asdf plugin add nodejs
+asdf plugin ad ruby
+
+# Install
 asdf install
 ```
 
-Start services with `docker-compose`:
+2. Then run database with docker:
 
+```bash
+docker compose up
 ```
-docker-compose up
+
+3. Run setup:
+
+```bash
+bin/setup
+```
+
+The project is now ready to run locally.
+
+## Run entirely project locally
+
+The steps are the same as for [Run locally using docker for database servers](#run-locally-using-docker-for-database-servers) except for step 2, and `docker` is (obviously) not a requirement.
+
+You will instead need to install the database servers and configure them correctly:
+
+- PostgreSQL (see version in docker-compose.yml)
+- Redis (see version in docker-compose.yml)
+
+It will be necessary to configure the database connections by setting corresponding env vars.
+If postgres is set up to use trust authentication (default for homebrew), then the following should be enough:
```

Comment:
Not sure people will know what is trust authentication maybe you should mention something if you are running `export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES` command regulary or encountering `__NSCFConstantString initialize` errors in logs

Reply:
I was thinking about making this line more `homebrew`-oriented.
So something like:
```md
If you've installed postgres and redis with homebrew (e.g. postgres is using trust authentication), then the following should be enough:
```

Reply:
For me it sounds the same as what is written, the issue is, that it does not tell me why should I set the DATABASE_SERVER to ""

What if I skipped this step because I thought that it was irrelevant for me, then I see errors and I would just run `export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES` as I am used to. 
This is just my perspective on how things would play out

Reply:
Is it better now?

Reply:
Chiming in here:

`OBJC_DISABLE_INITIALIZE_FORK_SAFETY` is not related in any way to Docker or Postgres. It is an issue with Ruby and MacOS.

Reply:
@davideluque Thank you for participating.

I'm aware that the issue is not really the template's or postgres' fault. Still it seems that our devs will likely face that specific issue, which is difficult to debug.

I am trying to be pragmatic here, without spending too much precious README space on it.

Let me know if you have a suggestion for how we can better solve this problem.

Reply:
Hey @jeppester 

I think it could be misleading to connect the NSFC error with the connection string / docker. This error is related to MacOS and Multithreading, as someone posted on StackOverflow:

> <img width="674" alt="Screenshot 2023-11-30 at 09 15 23" src="https://github.com/abtion/rails-template/assets/10492210/d09c0d5b-fded-489a-bd8a-f088aca7e9d3">

Reply:
Here's the link where someone experienced it in Python: https://stackoverflow.com/questions/50168647/multiprocessing-causes-python-to-crash-and-gives-an-error-may-have-been-in-progr

Reply:
The fact is that by changing the DATABASE_SERVER="" the error goes away

Reply:
To my best knowledge the situation is:
1. When using a socket to connect to a local postgres there is no issue
2. When using tcp/ip to connect to a local postgres the error happens

Emptying `DATABASE_SERVER` it will lead to 1, doing nothing will lead to 2.

What I'm trying to do here is to help our devs work around that problem rather having to figure out this illogical work-around by themselves.

What would you suggest that we do?